api/tag/janitor_test.go

```diff
@@ -14,12 +16,55 @@ func TestJanitor(t *testing.T) {
 	tagStore := NewMemoryStore()
 	taskProvider := mock_provider.NewMockTaskProvider(ctrl)
 
-	// todo: put some old tags in the store, mock the ListTasks call
+	tags := models.Tags{
+		{
+			EntityID:   "tsk_id1",
+			EntityType: "task",
+			Key:        "name",
+			Value:      "tsk_name1",
+		},
+		{
+			EntityID:   "tsk_id2",
+			EntityType: "task",
+			Key:        "name",
+			Value:      "tsk_name2",
+		},
+	}
+
+	for _, tag := range tags {
+		if err := tagStore.Insert(tag); err != nil {
+			t.Fatal(err)
+		}
+	}
+
+	tasks := []models.TaskSummary{
+		{
+			TaskID: "tsk_id2",
+		},
+	}
+
+	taskProvider.EXPECT().
+		List().
+		Return(tasks, nil)
 
 	janitor := NewJanitor(tagStore, taskProvider)
 	if err := janitor.Run(); err != nil {
 		t.Fatal(err)
 	}
 
-	// todo: assert only old tags got deleted
+	actual, err := tagStore.SelectByType("task")
```

Comment:
not sure why you're strafing away from using the existing `assert.Equal(t, expected, tagStore.tags)` pattern?

Reply:
Yup, you're right that is cleaner.